bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/crypto/EncryptionHandler.java

```diff
@@ -0,0 +1,307 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.communication.crypto;
+
+import java.util.Arrays;
+
+import org.openhab.binding.paradoxalarm.internal.util.ParadoxUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is used to encrypt and decrypt communication from/to Paradox system. Singleton pattern.
+ *
+ * Paradox encryption is using Rijndael 256-key expansion alghoritm.
+ * When key is changed the updateKey(byte[]) method needs to be called in front.
+ * Encrypt and Decrypt methods use the expandedKey field to do their job.
+ *
+ * The first packet sent to Paradox is the IP150 password as bytes, extended to 32 bytes with 0xEE.
+ * The first response contains the key that will be used for the rest of communication.
+ *
+ * Most of the coding is copy from Python / rewrite in Java from second link of jpbaracca's PAI repository. Probably
+ * some of the variables can be named better but I don't understand this code in it's full scope so I preferred to keep
+ * it as it is.
```

Comment:
Yeah, I'm not going to heed that. If you are trying to avoid IP issues it is best to change a much as possible.